src/main/java/com/chatcode/service/ScrapService.java

```diff
@@ -0,0 +1,54 @@
+package com.chatcode.service;
+
+import com.chatcode.domain.entity.Article;
+import com.chatcode.domain.entity.Avatar;
+import com.chatcode.domain.entity.Scrap;
+import com.chatcode.dto.ScrapResponseDto;
+import com.chatcode.exception.common.ContentNotFoundException;
+import com.chatcode.repository.article.ArticleWriteRepository;
+import com.chatcode.repository.avatar.AvatarWriteRepository;
+import com.chatcode.repository.scrap.ScrapReadRepository;
+import com.chatcode.repository.scrap.ScrapWriteRepository;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class ScrapService {
+    private final ScrapWriteRepository scrapWriteRepository;
+    private final ScrapReadRepository scrapReadRepository;
+    private final AvatarWriteRepository avatarWriteRepository;
+    private final ArticleWriteRepository articleWriteRepository;
+
+    @Transactional
+    public ScrapResponseDto scrap(Long articleId, Long avatarId) {
+        Avatar avatar = avatarWriteRepository.getReferenceById(avatarId);
+        Article article = articleWriteRepository.getReferenceById(articleId);
+
+        Scrap scrap = Scrap.builder()
+                .avatar(avatar)
+                .article(article)
+                .build();
+        scrapWriteRepository.save(scrap);
+
+        return ScrapResponseDto.of(scrap);
+    }
+
+    @Transactional
+    public List<ScrapResponseDto> getScrapList(Long avatarId) {
+        return scrapReadRepository.getScrapList(avatarId);
+    }
+
+    @Transactional
+    public void deleteScrap(Long articleId) {
+        Optional<Scrap> scrap = scrapWriteRepository.findByArticleId(articleId);
+        if (scrap.isPresent()) {
+            scrapWriteRepository.delete(scrap.get());
+            return;
+        }
+        throw new ContentNotFoundException("Scrap not found");
```

Comment:
```suggestion
        Scrap scrap = scrapWriteRepository.getReferenceById(articleId);
        scrapWriteRepository.delete(scrap);
```
여기 부분을 조금 짧게 줄여도 괜찮지 않을까 싶네요!
그리고 `findByArticleId` 자체는 직접 DB에서 entity를 조회해오는 거지만, `getReferenceById`는 DB를 조회하지 않고 영속성만 가지고 있는 거라, 실제 내부적으로는 delete by id로 삭제 될 겁니닷!

Reply:
앗 넵 !! getReferenceById를 쓰는 습관을 가져봐야겠어요 😂